File: crates/vikt-cli/tests/fixtures/folder/hub.js
// Cross-file hub, JavaScript side: jsHub does real work and calls jsHelper
// (its own file), and is called from wrapperA.js and wrapperB.js — two
// *different* files. Mirrors hub.py's shape exactly, so --scope repo has
// one same-language cross-file hub to rank in each of the two languages
// this folder mixes.

function jsHub(x) {
    let a = x + 1;
    let b = a * 2;
    let c = b - 3;
    return jsHelper(c) + a;
}

function jsHelper(x) {
    return x * 2;
}

File: crates/vikt-cli/tests/fixtures/folder/hub.py
"""Cross-file hub, Python side: py_hub does real work and calls py_helper (its
own file), and is called from wrapper_a.py and wrapper_b.py — two *different*
files. --scope file can only build a call graph within one file at a time, so
under file scope py_hub, wrapper_a and wrapper_b each look like a lone
function with no callers or callees; only --scope repo, which lets the call
graph and the re-rank both cross file boundaries, can see wrapper_a.py and
wrapper_b.py calling into this file at all.
"""


def py_hub(x):
    a = x + 1
    b = a * 2
    c = b - 3
    return py_helper(c) + a


def py_helper(x):
    return x * 2
